jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanReader.java

```diff
@@ -169,8 +171,18 @@ public Object read(JSONReader r, JsonParser p) throws IOException
                                 handleUnknown(r, p, propName);
                                 continue;
                             }
+                            props.add(propName);
                             values.add(prop.getReader().readNext(r, p));
                         }
+
+                        for(Field f : _valueType.getDeclaredFields()) {
```

Comment:
I don't think this is proper way to do this: we should fix the setting to find correct index, and as part of that process (or post-processing).
Or build data structures before actual read at any rate; not every time read occurs.